### PR TITLE
Suppress warnings in resolvectl about --type=

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Review by bluca — APPROVED** [review]
LGTM but not an expert on resolved

File: src/resolve/resolvectl.c [hunk 1]
```diff
@@ -476,7 +476,7 @@ static int resolve_record(sd_bus *bus, const char *name, uint16_t class, uint16_
 
         log_debug("Resolving %s %s %s (interface %s).", name, dns_class_to_string(class), dns_type_to_string(type), isempty(arg_ifname) ? "*" : arg_ifname);
 
-        if (single_label_nonsynthetic(name))
+        if (!endswith(name, ".") && single_label_nonsynthetic(name))
```
Comment (poettering) [inline src/resolve/resolvectl.c]: please use dns_name_dot_suffixed() for correctness (i.e. the trailing dot might be escaped in which case it isn't actually a dot)

Comment (keszybz) [reply src/resolve/resolvectl.c]: Thanks, updated.

File: src/resolve/resolvectl.c [hunk 2]
```diff
@@ -476,7 +476,7 @@ static int resolve_record(sd_bus *bus, const char *name, uint16_t class, uint16_
 
         log_debug("Resolving %s %s %s (interface %s).", name, dns_class_to_string(class), dns_type_to_string(type), isempty(arg_ifname) ? "*" : arg_ifname);
 
-        if (single_label_nonsynthetic(name))
+        if (!dns_name_dot_suffixed(name) && single_label_nonsynthetic(name))
```
Comment (poettering) [inline src/resolve/resolvectl.c]: this returns "int", not "bool", i.e. can fail. i.e. please acknowledge that in code, and do a <= 0 or == 0 or so

Comment (keszybz) [reply src/resolve/resolvectl.c]: Oh, indeed. Not sure how I missed that.

File: src/resolve/resolvectl.c [hunk 3]
```diff
@@ -476,7 +476,7 @@ static int resolve_record(sd_bus *bus, const char *name, uint16_t class, uint16_
 
         log_debug("Resolving %s %s %s (interface %s).", name, dns_class_to_string(class), dns_type_to_string(type), isempty(arg_ifname) ? "*" : arg_ifname);
 
-        if (single_label_nonsynthetic(name))
+        if (dns_name_dot_suffixed(name) > 0 && single_label_nonsynthetic(name))
```
Comment (poettering) [inline src/resolve/resolvectl.c]: don't you mean == 0?

Comment (keszybz) [reply src/resolve/resolvectl.c]: Probably ;)

**Pull request merged**